### ISSUE TITLE
Support introspection of materialized views

### ISSUE THREAD

**Issue by diogob** [kysely-org/kysely#270]
As I was investigating how to implement the [type generation for database views](https://github.com/RobinBlomberg/kysely-codegen/issues/53) I found out that the [MysqlIntrospector](https://github.com/koskimas/kysely/blob/master/src/dialect/mysql/mysql-introspector.ts) already returns views on a `getTables` call.

If all introspectors would behave in this way the feature would be automatically present in kysely-codegen.

Would there be any interest in merging a PR that makes postgres and sqlite `getTables` return database views?

**Comment by koskimas** [comment]
I didn't know `MysqlIntrospector` did that 😬

> Would there be any interest in merging a PR that makes postgres and sqlite getTables return database views?
Definitely! Combined with a `isView` property for `TableMetadata`.

**Comment by elmeister** [comment]
Thanks for your PR @diogob 👍 
Are there any plans to someday add support for materialized views as well?

**Comment by diogob** [comment]
@elmeister @koskimas the code change to support materialized views seems trivial. On the metadata interface front I believe no change would be necessary since a distinction between materialized view and view should be inconsequential to the application layer. We would simply report `isView = true` and treat them as any other view.

It would also require changes only on the PostgreSQL introspector since AFAIK there is no such feature in MySQL neither in SQLite.

@koskimas thoughts?

**Comment by igalklebanov** [comment]
I think we should support materialized views.

Seems like it would require additional round trip/s to get materialized view metadata, since version 9.3, they're not part of `information_schema`.
https://www.postgresql.org/docs/current/view-pg-matviews.html

If that's the case, I think that a good idea would be to extend `DatabaseMetadataOptions` with something like `withMaterializedViews: boolean = false`, to only make those trips when the consumer explicitly needs it.

**Comment by koskimas** [comment]
Yes, there's no reason to not support materialized views. I don't think we need a `withMaterializedViews` option either. We don't need additional round trips if we use joins/CTEs.
